pkg/reconciler/revision/resources/deploy.go

```diff
@@ -136,49 +135,63 @@ func makePodSpec(rev *v1.Revision, loggingConfig *logging.Config, tracingConfig
 	return podSpec, nil
 }
 
-// BuildUserContainer makes a container from the Revision template.
-func BuildUserContainer(rev *v1.Revision) *corev1.Container {
-	userContainer := rev.Spec.GetContainer().DeepCopy()
+// BuildUserContainers makes an array of containers from the Revision template.
+func BuildUserContainers(rev *v1.Revision) []corev1.Container {
+	containers := make([]corev1.Container, 0, len(rev.Spec.PodSpec.Containers))
+	for i := range rev.Spec.PodSpec.Containers {
+		var container corev1.Container
+		if len(rev.Spec.PodSpec.Containers[i].Ports) != 0 || len(rev.Spec.PodSpec.Containers) == 1 {
+			container = makeServingContainer(*rev.Spec.PodSpec.Containers[i].DeepCopy(), rev)
+		} else {
+			container = makeContainer(*rev.Spec.PodSpec.Containers[i].DeepCopy(), rev)
+		}
+		if rev.Status.ContainerStatuses != nil {
+			// Override container image with digest value if the ContainerStatuses contains non empty imageDigest value.
```

Comment:
Im not sure this comment really adds enough to be worth including tbh since it directly restates what the code says.

Reply:
Updated comment please have a look